C++/include/algorithm/dynamic_programming/coin_change.hpp

```diff
@@ -0,0 +1,65 @@
+/*
+    Coin Change Algorithm
+    ------------------------------------
+    Given an unlimited amount of coins with different values,
+    find the number of ways of making change for an given value using the coins.
+    This is a essential algorithm in dynamic programming that explores important
+    concepts.
+
+    Time complexity
+    ----------------
+    O(M*N), where M is the number of coins with different values and N is the amount that we desired to change
+
+    Space complexity
+    ----------------
+    O(M*N), where M is the number of coins with different values and N is the amount that we desired to change
+*/
+
+#ifndef COIN_CHANGE_HPP
+#define COIN_CHANGE_HPP
+
+#include <vector>
+using namespace std;

```

Comment:
Instead of `using namespace std`, write out the particular parts you're using, as in `using std::vector`. See https://stackoverflow.com/questions/1452721/why-is-using-namespace-std-considered-bad-practice